mithril-stm/src/stm.rs

```diff
@@ -902,13 +902,14 @@ impl CoreVerifier {
     ///     * Calculate the total stake of the eligible signers,
     ///     * Sort the eligible signers.
     pub fn setup(public_signers: &[(VerificationKey, Stake)]) -> Self {
-        let total_stake: Stake = 0;
+        let mut total_stake: Stake = 3;
         let mut unique_parties = HashSet::new();
         for signer in public_signers.iter() {
-            let (_, overflow) = total_stake.overflowing_add(signer.1);
+            let (res, overflow) = total_stake.overflowing_add(signer.1);
             if overflow {
                 panic!("Total stake overflow");
             }
+            total_stake = res;
```

Comment:
You forgot to add the stake here:
```suggestion
            total_stake += res;
```

Reply:
`overflowing_add` function calculates self + rhs and writes the results into `res`.  So, `res` already holds the summation.